tests/core/test_expectation_suite.py

```diff
@@ -26,7 +33,290 @@ def exp1():
+            evaluation_parameters=test_evaluation_parameters,
+            data_asset_type=test_data_asset_type,
+            execution_engine_type=dummy_execution_engine_type,  # type: ignore[arg-type]
+            meta=test_meta,
+            ge_cloud_id=test_id,
+        )
+        assert suite.expectation_suite_name == fake_expectation_suite_name
+        assert suite._data_context == dummy_data_context
+        assert suite.expectations == [expect_column_values_to_be_in_set_col_a_with_meta]
+        assert suite.evaluation_parameters == test_evaluation_parameters
+        assert suite.data_asset_type == test_data_asset_type
+        assert suite.execution_engine_type == dummy_execution_engine_type
+        assert suite.meta == test_meta
+        assert suite.ge_cloud_id == test_id
+
+    @pytest.mark.unit
+    def test_expectation_suite_init_overrides_expectations_dict_and_obj(
+        self,
+        fake_expectation_suite_name: str,
+        expect_column_values_to_be_in_set_col_a_with_meta_dict: dict,
+        expect_column_values_to_be_in_set_col_a_with_meta: ExpectationConfiguration,
+    ):
+        """What does this test and why?
+
+        The expectations param of ExpectationSuite takes a list of ExpectationConfiguration or dicts and both can be provided at the same time. We need to make sure they both show up as expectation configurations in the instantiated ExpectationSuite.
+        """
+
+        test_expectations_input = [
+            expect_column_values_to_be_in_set_col_a_with_meta_dict,
+            expect_column_values_to_be_in_set_col_a_with_meta,
+        ]
+
+        suite = ExpectationSuite(
+            expectation_suite_name=fake_expectation_suite_name,
+            expectations=test_expectations_input,  # type: ignore[arg-type]
+        )
+        assert suite.expectation_suite_name == fake_expectation_suite_name
+
+        test_expected_expectations = [
+            ExpectationConfiguration(
+                **expect_column_values_to_be_in_set_col_a_with_meta_dict
+            ),
+            expect_column_values_to_be_in_set_col_a_with_meta,
+        ]
+        assert len(suite.expectations) == 2
+        assert suite.expectations == test_expected_expectations
+
+    @pytest.mark.unit
+    def test_expectation_suite_init_overrides_non_json_serializable_meta(
+        self,
+        fake_expectation_suite_name: str,
+    ):
+        """What does this test and why?
+
+        meta field overrides need to be json serializable, if not we raise an exception.
+        """
+
+        class NotSerializable:
+            def __dict__(self):
+                raise NotImplementedError
+
+        test_meta = {"this_is_not_json_serializable": NotSerializable()}
+
+        with pytest.raises(InvalidExpectationConfigurationError):
+            ExpectationSuite(
+                expectation_suite_name=fake_expectation_suite_name,
+                meta=test_meta,  # type: ignore[arg-type]
+            )
```

Comment:
could we add `as e` and `assert "blah" in str(e.value)` for any `pytest.raises` calls around error we raise?

Reply:
Good call! I only added for the other one in this PR. Adding now.